Example/Messaging/Sample/Sample/ContentView.swift

```diff
@@ -0,0 +1,143 @@
+// Copyright 2020 Google
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import FirebaseCore
+import FirebaseMessaging
+import FirebaseInstanceID
+import FirebaseInstallations
+
+struct ContentView: View {
+  @EnvironmentObject var identity: Identity
+
+  var body: some View {
+    NavigationView {
+      List {
+        Button(action: getToken) {
+          HStack {
+            Image(systemName: "arrow.clockwise.circle.fill").font(.body)
+            Text("Get ID and Token")
+              .fontWeight(.semibold)
+          }
+          .padding(5)
+          .foregroundColor(.white)
+          .background(Color.purple)
+          .cornerRadius(40)
+        }
```

Comment:
These look really nice - just a few suggestions here.

There's a lot of duplication here and below, there are two ways I think this could be resolved:

1. Creating a custom button `struct PillButton` or something, that takes in the `title`, `systemImageName`, `backgroundColor`, and `action`. That way the actual layout is only in one place and you can modify it once for all buttons. I think a nice API would be like:

```
PillButton(title: "Get ID and Token",
           systemImageName:"arrow.clockwise.circle.fill",
           backgroundColor: .purple,
           action: getToken)
```
.. and then if it's a short action, it could be added inline:

```
PillButton(title: "Say Hello",
           systemImageName:"arrow.clockwise.circle.fill",
           backgroundColor: .purple) {
  print("Hello!")
}
```

Let me know if you need a hand creating it!

2. You could use a custom [`ButtonStyle`](https://developer.apple.com/documentation/swiftui/buttonstyle), which is really nice. Could let you do something like:

```
struct PillButton: ButtonStyle {
  // implementation...
}

// Usage:
Button("Get ID and Token", action: getToken)
  .buttonStyle(PillButton(icon: "arrow.clockwise.circle.fill", background: .purple))
```

Part of the problem with 2 is you lose the iOS provided pressing animation (going darker to indicate you've pressed it down). You can customize it since the ButtonStyle has a configuration that tells you when it's pressed, but it's more for really custom buttons that wouldn't act like a traditional button (scales when it's pressed, something else).

Reply:
Secondly, for the padding it's probably reasonable to keep the default value: `.padding()` that way it's a large enough touch target and it'll choose the right padding based on the platform (just a good practice, I know this isn't on other platforms).

Reply:
nice!